sql/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2793,14 +2804,12 @@ class SparkConnectPlanner(
         .build()
     }
 
-    val df = relation.getRelTypeCase match {
-      case proto.Relation.RelTypeCase.SQL =>
-        executeSQL(relation.getSql, tracker)
-      case proto.Relation.RelTypeCase.WITH_RELATIONS =>
-        executeSQLWithRefs(relation.getWithRelations, tracker)
-      case other =>
-        throw InvalidInputErrors.sqlCommandExpectsSqlOrWithRelations(other)
+    // Only allow a SQL relation or a SQL relation nested in a WithRelations relation.
+    if (!relation.hasSql &&
+      !(relation.hasWithRelations && relation.getWithRelations.getRoot.hasSql)) {
+      throw InvalidInputErrors.sqlCommandExpectsSqlOrWithRelations(relation.getRelTypeCase)
     }
+    val df = Dataset.ofRows(session, transformRelation(relation), tracker)
```

Comment:
Capture the parse time here, use the transformRelation time as a proxy. Long term we may want to look at the use of QueryPlanTracker in general.